app/controllers/chopps_controller.rb

```diff
@@ -0,0 +1,20 @@
+class ChoppsController < ApplicationController
+  skip_before_action :authenticate_request
+
+  def validate_qrcode
+    chopp = Chopp.find_by(qrcode: params[:qrcode])
+    if chopp && chopp.qrcode_validate
+      chopp_attr = {:size => chopp.size, :chopp_type => chopp.chopp_type, :collar => chopp.collar}
+      chopps_combinations = Chopp.create_chopps_combinations
```

Comment:
Pelo que entendi, se der certo você retorna a combinação do chopp, precisa documentar essas combinações para quem for receber essa validação saber o que significa.

Reply:
Recomendo criar/atualizar nossos requisitos para o pessoal da eletrônica. @Oziel-Silva  e @ithallojunior, vocês já sabem como vão liberar o chopp através desse pedido?

Reply:
Eu montei a tabelinha, é basicamente assim: 

```
0- {:size=>500, :chopp_type=>"tradicional", :collar=>0}
1- {:size=>500, :chopp_type=>"tradicional", :collar=>1}
2- {:size=>500, :chopp_type=>"tradicional", :collar=>2}
3- {:size=>500, :chopp_type=>"vinho", :collar=>0}
4- {:size=>500, :chopp_type=>"vinho", :collar=>1}
5- {:size=>500, :chopp_type=>"vinho", :collar=>2}
6- {:size=>1000, :chopp_type=>"tradicional", :collar=>0}
7- {:size=>1000, :chopp_type=>"tradicional", :collar=>1}
8- {:size=>1000, :chopp_type=>"tradicional", :collar=>2}
9- {:size=>1000, :chopp_type=>"vinho", :collar=>0}
10- {:size=>1000, :chopp_type=>"vinho", :collar=>1}
11- {:size=>1000, :chopp_type=>"vinho", :collar=>2}

```

O método já retorna o código, logo se retornar `0` é referente a combinação: `  {:size=>500, :chopp_type=>"tradicional", :collar=>0}`